### DISCUSSION TITLE
Request compression Unable to submit

### DISCUSSION THREAD

**Discussion by bearfishops** [prometheus/pushgateway#511] [Q&A] [answered]
I run this shell command。
echo "some_metric 3.14" | gzip | curl --data-binary @- http://xxx.xxx.xxx.xxx:59091/metrics/job/xy_influxdb_exporter/instance/test1

Unable to submit normally. The following information is returned。
 text format parsing error in line 1: invalid metric name
![image](https://user-images.githubusercontent.com/117809213/200772373-d62a5c10-1ad0-4a10-b93c-122806b144ca.png)

**Comment by 1skt0** [comment]
i have the same problem

**Comment by beorn7** [comment]
You have to add the `Content-Encoding: gzip` header, as mentioned in the example:

```
echo "some_metric 3.14" | gzip | curl -H 'Content-Encoding: gzip' --data-binary @- http://pushgateway.example.org:9091/metrics/job/some_job
```

**Reply by gaoyanyu** [reply]
problem exists even "Content-Encoding: gzip" added to header, like this:
![企业微信截图_16691134114200](https://user-images.githubusercontent.com/32916782/203292586-0b585020-ed4d-4b0e-a62c-2cb5f7cd44bb.png)

**Comment by beorn7** [comment]
In this case, I assume you are using one of the released binaries, while the feature is not yet in any release (just in the master branch).

I do plan to cut a release very soon.

**Comment by beorn7** [comment, marked as answer]
The new release is out: https://github.com/prometheus/pushgateway/releases/tag/v1.5.0

**Reply by bearfishops** [reply]
Thank you. After I switch the release to v1.5.0, it can be submitted successfully

**Comment by tgeo-cambrian** [comment]
@beorn7 Hello, does it need an extra flag or something when launching the container to accept the gzip output? 

i.e. I face the below issue in version 1.5.0 
![image](https://user-images.githubusercontent.com/113511859/204546793-067a602f-a2f6-43ae-9de7-9c482a9ddfed.png)

I am pushing some metrics from a windows ephemeral VM
first getting the metrics from wmi exporter

```
curl http://localhost:9182/metrics
```

gziping them
```
tar -czvf metricscmd.gz metricscmd
```

```H:\My Desktop>curl -H Content-Encoding:gzip --data-binary @metricscmd.gz http://<redacted>:9091/metrics/job/test-job/instance/test-curl-cmd -vvvv
*   Trying <redacted>:9091...
* Connected to <redacted> port 9091 (#0)
> POST /metrics/job/test-job/instance/test-curl-cmd HTTP/1.1
> Host: <redacted>:9091
> User-Agent: curl/7.83.1
> Accept: */*
> Content-Encoding:gzip
> Content-Length: 68950
> Content-Type: application/x-www-form-urlencoded
>
* We are completely uploaded and fine
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Date: Tue, 29 Nov 2022 15:07:20 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 435
< Connection: keep-alive
< X-Content-Type-Options: nosniff
<
text format parsing error in line 1: expected float as value, got "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00000666"
* Connection #0 to host <redacted> left intact```

**Reply by beorn7** [reply]
I would assume that `tar`ing the payload adds additional data to it that cannot be read by the Pushgateway.

You have to just gzip the metrics.

In different news, the Pushgateway is not designed to receive the metrics from a regular Prometheus endpoint. You should scrape that endpoint directly, maybe using something like https://github.com/prometheus-community/PushProx if you cannot reach it directly.

**Reply by tgeo-cambrian** [reply]
@beorn7 I know, the design decision is not mine. You are right tar does some funky stuff, and it needs gzip or 7z gzip implementation which is clear. I can confirm it works, just need to work around windows issues. Thanks alot.
